maths/harshad_numbers.py

```diff
@@ -0,0 +1,132 @@
+"""
+A Harshad number is divisible by the sum of its digits in any base n.
+Reference: https://en.wikipedia.org/wiki/Harshad_number
+"""
+
+
+def int_to_base(number: int, base_of_interest: int) -> str:
+    digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    result = ""
+
+    while number > 0:
+        number, remainder = divmod(number, base_of_interest)
+        result = digits[remainder] + result
+
+    if result == "":
+        result = "0"
+
+    return result
+
+
+def sum_of_digits(num: int, base_of_interest: int) -> str:
+    """
+    Calculate the sum of digit values in a positive integer
+    converted to the given 'base_of_interest'.
+    Where 'base_of_interest' ranges from 2 to 36.
+
+    Examples:
+    >>> sum_of_digits(103, 12)
+    '13'
+    >>> sum_of_digits(1275, 4)
+    '30'
+    >>> sum_of_digits(6645, 2)
+    '1001'
+    >>> # bases beyond 36 and below 2 will error
+    >>> sum_of_digits(543, 1)
+    Traceback (most recent call last):
+        ...
+    ValueError: 'base_of_interest' must be between 36 and 2 inclusive
+    >>> sum_of_digits(543, 37)
+    Traceback (most recent call last):
+        ...
+    ValueError: 'base_of_interest' must be between 36 and 2 inclusive
+    """
+
+    if (base_of_interest > 36) or (base_of_interest < 2):
+        raise ValueError("'base_of_interest' must be between 36 and 2 inclusive")
+
+    num_str = int_to_base(num, base_of_interest)
+    res = 0
+    for char in num_str:
+        res += int(char, base_of_interest)
+    res_str = int_to_base(res, base_of_interest)
+    return res_str
+
+
+def all_harshad_numbers(num: int, base_of_interest: int) -> tuple[int, list[str]]:

```

Comment:
```suggestion
def harshad_numbers_in_base(limit: int, base: int) -> list[str]:
```
1) Renamed function and variables to be more accurate
2) Just return the list of harshad numbers, the user can easily get the number of harshad numbers by just getting the length of the list